tools/adventure-pack/goodies/typescript/Iterator.prototype.every/index.ts

```diff
@@ -0,0 +1,25 @@
+import "../Iterator.prototype.toIterable";
+import { iteratorPrototype } from "../Iterator.prototype";
+
+declare global {
+  interface Iterator<T> {
+    every(
+      this: Iterator<T>,
+      callbackfn: (value: T, index: number) => unknown,
+    ): boolean;
+  }
+}
+
+iteratorPrototype.every ??= function <T>(
+  this: Iterator<T>,
+  callbackfn: (element: T, index: number) => unknown,
+): boolean {
+  let index = 0;
+  for (const element of this.toIterable()) {
+    if (!callbackfn(element, index)) {
+      return false;
```

Comment:
It would be fun to have a test that asserts that we didn't check every element to determine the answer!

Reply:
ack!